src/fivetran-destination/src/logging.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use serde::Serialize;
+use tracing_core::{Event, Level, Subscriber};
+use tracing_subscriber::fmt::format::Writer;
+use tracing_subscriber::fmt::{format, FmtContext, FormatEvent, FormatFields};
+use tracing_subscriber::registry::LookupSpan;
+
+use std::io;
+
+/// Implements [`tracing_subscriber::fmt::FormatEvent`] according to the Fivetran SDK logging
+/// format.
+///
+/// See: <https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging>
+pub struct FivetranLoggingFormat {
+    origin: FivetranEventOrigin,
+}
+
+impl FivetranLoggingFormat {
+    /// Returns a type that implements [`FormatEvent`] and is configured to log for a Fivetran
+    /// Destination.
+    pub fn destination() -> Self {
+        FivetranLoggingFormat {
+            origin: FivetranEventOrigin::SdkDestination,
+        }
+    }
+}
+
+impl<S, N> FormatEvent<S, N> for FivetranLoggingFormat
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'a> FormatFields<'a> + 'static,
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        mut writer: format::Writer<'_>,
+        event: &Event<'_>,
+    ) -> std::fmt::Result {
+        let mut message = String::new();
+        ctx.format_fields(Writer::new(&mut message), event)?;
```

Comment:
what does this do?

Reply:
AFAIU a single event doesn't necessarily contain all of the "fields" it has associated with it, and you need to use a visitor pattern to properly format them. Using `format_fields` from the `FmtContext` does this.

It's also what the implements of `FormatEvent` from `tracing_subscriber` do, [example](https://github.com/tokio-rs/tracing/blob/master/tracing-subscriber/src/fmt/format/mod.rs#L1034)